packages/api/src/routers/svc/user.ts

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import cors from 'cors'
+import express from 'express'
+import { LessThan } from 'typeorm'
+import { StatusType } from '../../entity/user'
+import { readPushSubscription } from '../../pubsub'
+import { deleteUsers } from '../../services/user'
+import { corsConfig } from '../../utils/corsConfig'
+import { logger } from '../../utils/logger'
+
+type CleanupMessage = {
+  subDays: number
+}
+
+const isCleanupMessage = (obj: any): obj is CleanupMessage =>
+  'subDays' in obj && !isNaN(obj.subDays)
+
+const getCleanupMessage = (msgStr: string): CleanupMessage => {
+  try {
+    const obj = JSON.parse(msgStr) as unknown
+    if (isCleanupMessage(obj)) {
+      return obj
+    }
+  } catch (err) {
+    console.log('error deserializing event: ', { msgStr, err })
+  }
+
+  return {
+    subDays: 0, // default to 0
+  }
+}
+
+export function userServiceRouter() {
+  const router = express.Router()
+
+  router.post('/prune', cors<express.Request>(corsConfig), async (req, res) => {
+    logger.info('prune soft deleted users')
+
+    const { message: msgStr, expired } = readPushSubscription(req)
```

Comment:
This function also validates the token, right?

Reply:
Yeah